doc/plugins/hmac-auth.md

```diff
@@ -205,3 +209,201 @@ $ curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f
     }
 }'
 ```
+
+## Generate Signature Examples
+
+Take HMAC SHA256 as an example to introduce the signature generation examples in different languages.
+
+Example inputs:
+Variable | Value
+---|---
+secret | this is secret key
+message | this is signature string
+
+Example outputs:
+Type | Hash
+---|---
+hexit | ad1b76c7e5054009380edca35d3f36cc5b6f45c82ee02ea3af64197ebddb9345
+base64 | rRt2x+UFQAk4DtyjXT82zFtvRcgu4C6jr2QZfr3bk0U=
+
+### Python 3
```

Comment:
Maybe we can move the code examples to a separate file since they are too big.

Reply:
I think these code examples are good in this file, so it is more intuitive. If the document is placed in a different location, it is easy for users to miss it.

Reply:
I considered it, but I did not find a suitable folder to put this file. 
Do you have any good suggestions?
Looking forward to your suggestions.

Reply:
@fukiki 
What about making a new directory `doc/examples`, put file there, and leave a link in this plugin's doc?

Reply:
@Firstsawyou Although these code examples are more intuitive in this file, it's better to maintain in a separate file.
@spacewander Thanks for your suggestion!
I modified my PR